torch_xla/runtime.py

```diff
@@ -44,10 +45,15 @@ def _maybe_select_default_device():
   if torch_xla._found_libtpu and tpu.num_available_chips() > 0:
     logging.warning('libtpu.so and TPU device found. Setting PJRT_DEVICE=TPU.')
     os.environ[xenv.PJRT_DEVICE] = 'TPU'
-  # TODO(wcromar): Detect GPU device
   elif xu.getenv_as(xenv.GPU_NUM_DEVICES, int, 0) > 0:
     logging.warning('GPU_NUM_DEVICES is set. Setting PJRT_DEVICE=CUDA')
     os.environ[xenv.PJRT_DEVICE] = 'CUDA'
+  elif torch.cuda.is_available():
+    logging.warning(
+        'Found CUDA without GPU_NUM_DEVICES. Defaulting to PJRT_DEVICE=CUDA with GPU_NUM_DEVICES=1.'
+    )
+    os.environ[xenv.PJRT_DEVICE] = 'CUDA'
+    os.environ[xenv.GPU_NUM_DEVICES] = '1'
```

Comment:
how do you know GPU_NUM_DEVICES is 1?

Reply:
Good point, I didn't :). I just defaulted to 1 naively. Changed to default to the number of devices torch detected.